x-pack/plugins/reporting/public/plugin.ts

```diff
@@ -245,6 +247,7 @@ export class ReportingPublicPlugin
             application,
             usesUiCapabilities,
             theme: core.theme,
+            enable: !this.config.export_types.pdf.enabled || !this.config.export_types.png.enabled,
```

Comment:
I'd like to suggest this, it will allow us to not have to update the share provider functions:
```typescript
        if (this.config.export_types.pdf.enabled || this.config.export_types.png.enabled) {
          share.register(
            reportingScreenshotShareProvider({
              apiClient,
              toasts,
              uiSettings,
              license,
              application,
              usesUiCapabilities,
              theme: core.theme,
            })
          );
        }
```